scripts/lib/CIME/XML/env_batch.py

```diff
@@ -167,26 +167,34 @@ def set_batch_system(self, batchobj, batch_system_type=None):
     def make_batch_script(self, input_template, job, case, outfile=None):
         expect(os.path.exists(input_template), "input file '{}' does not exist".format(input_template))
         env_workflow = case.get_env('workflow')
-        task_count = env_workflow.get_value("task_count", subgroup=job)
-        tasks_per_node = env_workflow.get_value("tasks_per_node", subgroup=job)
-        if not tasks_per_node:
-            tasks_per_node = case.tasks_per_node
+        total_tasks, num_nodes, tasks_per_node, thread_count = env_workflow.get_job_specs(job)
```

Comment:
allow jobs other than case.run and case.test to use mpirun, this was neglected in the past because the only other job in the original workflow case.st_archive does not use mpirun.